src/main.cc

```diff
@@ -1411,15 +1417,17 @@ SquidMainSafe(int argc, char **argv)
 
 /// computes name and ID for the current kid process
 static void
-ConfigureCurrentKid(const char *processName)
+ConfigureCurrentKid(const CommandLine &cmdLine)
 {
-    // kids are marked with parenthesis around their process names
-    if (processName && processName[0] == '(') {
+    const char *kidParams = nullptr;
+    if (cmdLine.hasOption(optKid, &kidParams)) {
+        // TODO: Use Tokenizer to parse kidParams.
+        char *processName = xstrdup(kidParams);
         if (const char *idStart = strrchr(processName, '-')) {
             KidIdentifier = atoi(idStart + 1);
-            const size_t nameLen = idStart - (processName + 1);
+            const size_t nameLen = idStart - processName;
             assert(nameLen < sizeof(TheKidName));
-            xstrncpy(TheKidName, processName + 1, nameLen + 1);
+            xstrncpy(TheKidName, processName, nameLen);
```

Comment:
Here is a bug: xstrncpy() expects nameLen+1, not nameLen. Addressing the above TODO should fix this also.

Reply:
small correction: xstrncpy() requires the amount of characters to be copied including the nul-terminator. IF (and only if) the nameLen value excludes terminator a +1 is required.

Reply:
This is why we should be using a tokenizer...

Reply:
Done with the tokenizer.